### PR TITLE
Add support for converting images to stargzify

### PR DESCRIPTION
**Pull request by jonjohnsonjr**
A version of stargzify that works against container registries.

We probably want a better name and different flags, but this is a start. We also might want to make layer flattening optional?

### CODE REVIEW

**Comment by bradfitz** [comment]
Drop the go.mod and go.sum files. We'll just have one repo-wide module.

Can you just put this new code in the existing stargzify command? Feel free to change its flags and usage.

**Comment by jonjohnsonjr** [comment]
Replaced stargzify with what I had and made it positional-args based instead of flag-based.

File: stargz/registargzify/registargzify.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+var (
+	in  = flag.String("in", "", "image to be stargzified.")
+	out = flag.String("out", "", "destination for that image.")
+)
+
+func main() {
+	flag.Parse()
+	src, dst := *in, *out
+	srcRef, err := name.ParseReference(src, name.WeakValidation)
```
Comment (bradfitz) [inline stargz/registargzify/registargzify.go]: Will this still support local files on disk?

File: stargz/registargzify/registargzify.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/empty"
+	"github.com/google/go-containerregistry/pkg/v1/mutate"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/google/go-containerregistry/pkg/v1/tarball"
+)
+
+var (
+	in  = flag.String("in", "", "image to be stargzified.")
```
Comment (bradfitz) [inline stargz/registargzify/registargzify.go]: Clarify what forms of input this accepts? Local files? gcr.io URLs? Stdin?

File: stargz/registargzify/registargzify.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	out = flag.String("out", "", "destination for that image.")
```
Comment (bradfitz) [inline stargz/registargzify/registargzify.go]: likewise

File: stargz/registargzify/registargzify.go [hunk 4]
```diff
@@ -0,0 +1,105 @@
+// Copyright 2019 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// The stargzify command converts a tarball into a seekable stargz
```
Comment (bradfitz) [inline stargz/registargzify/registargzify.go]: The package comment will need updating.

**Comment by bradfitz** [comment]
@jonjohnsonjr, you got any time to finish this up?

File: stargz/registargzify/registargzify.go (hunk 1 above)
Comment (jonjohnsonjr) [reply stargz/registargzify/registargzify.go]: I could support both files and images but I'm struggling to find a flag/argument scheme that makes using the tool not awkward. That's why I originally had it as a separate tool.

**Comment by jonjohnsonjr** [comment]
Added file conversion back when invoked with an argument prefixed by `file:`. Once https://github.com/google/go-containerregistry/pull/426 lands I can use https://github.com/google/crfs/pull/8 to do the upload in one pass instead of writing out the layer to a temp file.

File: stargz/stargzify/stargzify.go [hunk 5]
```diff
@@ -25,11 +27,121 @@ import (
...
+	src = args[0]
+
+	if len(args) < 2 {
+		if *upgrade {
+			dst = src
+		} else {
+			printUsage()
+		}
+	} else {
+		dst = args[1]
```
Comment (bradfitz) [inline stargz/stargzify/stargzify.go]: You should also give a usage message if --upgrade is used with len(args) == 2

and complain in general if len(args) > 2

**Comment by jonjohnsonjr** [comment]
Updated error messages and added a `-flatten` flag.

**Comment by bradfitz** [comment]
LGTM.

**Pull request merged**